packages/fx-core/src/component/driver/middleware/addSpecialScriptTelemetry.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { HookContext, Middleware, NextFunction } from "@feathersjs/hooks";
+import { WrapDriverContext } from "../util/wrapUtil";
+import {
+  TeamsFxTelemetryConfig,
+  TeamsFxTelemetryReporter,
+} from "../../utils/teamsFxTelemetryReporter";
+import { TelemetryConstant } from "../../constant/commonConstant";
+import { performance } from "perf_hooks";
+import { TelemetryConstants } from "../../constants";
+import { isExecutionResult } from "./addStartAndEndTelemetry";
+import { maskSecretValues } from "../../utils/envUtil";
+
+/**
+ * A special telemetry middleware for script driver running for a SWA deploy.
+ * @param eventName The event name to be sent.
+ */
+export function addSpecialScriptTelemetry(eventName: string): Middleware {
+  return async (ctx: HookContext, next: NextFunction) => {
+    const name = ctx.arguments[4] as string | undefined;
+    const command = maskSecretValues(ctx.arguments[0].run as string);
+    // only add telemetry for script
+    if (name?.indexOf("deploy to Azure Static Web Apps") ?? -1 >= 0) {
```

Comment:
Is this "?? -1" needed?

Reply:
Sure `name?.indexOf("deploy to Azure Static Web Apps")` equals undefined when the name is undefined and cannot compare with 0.